### PR TITLE
Menu bar Items

### PR DESCRIPTION
**Pull request by SirD-AL**
Overall of the kind variable, will keep cleaning the files but you can review this one before Thursday.
I will update it more before then.

### CODE REVIEW

File: src/model.rs [hunk 1]
```diff
@@ -501,20 +507,139 @@ impl Component for Model {
...
+                    let parent = coord.parent().unwrap();
+
+
+                    let mut temp : Vec<Grammar> = vec![];
+                    let mut u = 0;
+                    let mut row_coords2 = self.query_row(next_row.full_row());
+                    let mut new_sub_coords : std::vec::Vec<(std::num::NonZeroU32, std::num::NonZeroU32)> = vec![];
+                    let parent = coord.parent().unwrap();
+                    //Keeping track for grid in progress
+                    // if let Some(Grammar{ kind: Kind::Grid(sub_coords), name:_, style:_ }) = self.grammars.get(&parent) {
```
Comment (Korede-TA) [inline src/model.rs]: if these comments are no longer relevant, please delete

File: src/model.rs [hunk 2]
```diff
@@ -501,20 +507,139 @@ impl Component for Model {
...
+                    let parent = coord.parent().unwrap();
+
+
+                    let mut temp : Vec<Grammar> = vec![];
+                    let mut u = 0;
+                    let mut col_coords2 = self.query_col(next_col.full_col());
+                    let mut new_sub_coords : std::vec::Vec<(std::num::NonZeroU32, std::num::NonZeroU32)> = vec![];
+                    let parent = coord.parent().unwrap();
+                    //Keeping track for grid in progress
+                    // if let Some(Grammar{ kind: Kind::Grid(sub_coords), name:_, style:_ }) = self.grammars.get(&parent) {
```
Comment (Korede-TA) [inline src/model.rs]: remove obsolete comments

File: src/model.rs (hunk 1 above)
Comment (SirD-AL) [reply src/model.rs]: Yeah will clean up

**Comment by SirD-AL** [comment]
@Korede-TA In terms row manipulation, the thing is here we are keeping track of the number of rows and columns which is process to be a grid when being rendered rather than specifying the grid layout at each modification manually. I do need to check on the values that are stored but in terms of coordinates, after printing (Info!) them there isn't any problem with their changes.

For now as we started it is only taking care of the last row or column, in terms of values even tho the console says otherwise. However I do believe that having the coordinates be dynamic is the best solution as grammars changes in terms of style of value, we want to be able to modify those coordinates easily without having to input a new grid hash map to its parent and keeping track of all the Childs coordinates asweell.

So here the grid/kind keeps the number of rows and columns, and at each serenader it reassigns the coordinates by using the macro (get_grid).

**Comment by SirD-AL** [comment]
@Korede-TA Now the deletes are working good ready to merge, Deletes shift the grammar one row or column in a loop then deletes the last one. 
Please test it out. 
I kept the info! commented for troubleshooting as I am trying many situations.

**Pull request merged**